File: src/elixir/svd/lib/svd/models/port_information.ex
defmodule SVD.Models.PortInformation do
  @moduledoc false

  @fields [
    :departure_port_code,
    :departure_port_description,
    :departure_port_name,
    :arrival_port_code,
    :arrival_port_description,
    :arrival_port_name,
    :inbound_port_jurisdiction_code,
    :outbound_port_jurisdiction_code,
    :pilot_boarding_place_name,
    :pilot_boarding_place_location,
    :berth_name
  ]

  @derive {Jason.Encoder, only: @fields}
  defstruct @fields

  @type t :: %__MODULE__{}

  @spec fields() :: [atom()]
  def fields, do: @fields
end

File: src/go/svd/models/port_information.go
package models

// PortInformation represents the "Port and Route" section of SVD.
type PortInformation struct {
	DeparturePortCode            string `json:"departurePortCode" xml:"DeparturePortCode" csv:"DeparturePortCode"`
	DeparturePortName            string `json:"departurePortName" xml:"DeparturePortName" csv:"DeparturePortName"`
	ArrivalPortCode              string `json:"arrivalPortCode" xml:"ArrivalPortCode" csv:"ArrivalPortCode"`
	ArrivalPortName              string `json:"arrivalPortName" xml:"ArrivalPortName" csv:"ArrivalPortName"`
	InboundPortJurisdictionCode  string `json:"inboundPortJurisdictionCode" xml:"InboundPortJurisdictionCode" csv:"InboundPortJurisdictionCode"`
	OutboundPortJurisdictionCode string `json:"outboundPortJurisdictionCode" xml:"OutboundPortJurisdictionCode" csv:"OutboundPortJurisdictionCode"`
	PilotBoardingPlaceName       string `json:"pilotBoardingPlaceName" xml:"PilotBoardingPlaceName" csv:"PilotBoardingPlaceName"`
	PilotBoardingPlaceLocation   string `json:"pilotBoardingPlaceLocation" xml:"PilotBoardingPlaceLocation" csv:"PilotBoardingPlaceLocation"`
	BerthName                    string `json:"berthName" xml:"BerthName" csv:"BerthName"`
}
